test/objects.coffee

```diff
@@ -582,6 +582,194 @@ test "#4324: Shorthand after interpolated key", ->
   eq 1, obj[1]
   eq 2, obj.a
 
+test "object computed property keys", ->
+  # Simple cases.
+  a = 'a'
+  obj = [a]: yes
+  eq obj.a, yes
+  obj = {[a]: yes}
+  eq obj.a, yes
+  obj = {[a]}
+  eq obj.a, 'a'
+  obj = {[5]}
+  eq obj[5], 5
+  obj = {['5']}
+  eq obj['5'], '5'
+
+  # Commas in implicit object.
```

Comment:
Please split up each section into a new `test`. Long tests are annoying, for if they fail there are lots of places to check to find where exactly the failure is. Ideally each `test` function would have only one validation (e.g. only one `eq`).

Reply:
Agree. I copied the whole block from "object keys with interpolations".
I'll split them into smaller sections.